src/algebra/iterate_hom.lean

```diff
@@ -128,18 +129,71 @@ end ring_hom
 lemma equiv.perm.coe_pow {α : Type*} (f : equiv.perm α) (n : ℕ) : ⇑(f ^ n) = (f^[n]) :=
 hom_coe_pow _ rfl (λ _ _, rfl) _ _
 
-@[simp] lemma mul_left_iterate [monoid M] (a : M) (n : ℕ) : ((*) a)^[n] = (*) (a^n) :=
+--what should be the namespace for this section?
+section monoid
+
+variables [monoid G] {a : G} (n : ℕ)
+
+variable (a)
+@[simp] lemma mul_left_iterate : ((*) a)^[n] = (*) (a^n) :=
 nat.rec_on n (funext $ λ x, by simp) $ λ n ihn,
 funext $ λ x, by simp [iterate_succ, ihn, pow_succ', mul_assoc]
 
-@[simp] lemma add_left_iterate [add_monoid M] (a : M) (n : ℕ) : ((+) a)^[n] = (+) (n •ℕ a) :=
+@[simp] lemma mul_right_iterate : (* a)^[n] = (* a ^ n) :=
+begin
+  induction n with d hd,
+  { simpa },
+  { simpa [hd] }
+end
+
+lemma mul_right_iterate_apply_one : (* a)^[n] 1 = a ^ n :=
+by simp [mul_right_iterate]
+
+end monoid
+
+section semigroup
+
+variables [semigroup G] {a b c : G}
+
+variable {a}
+@[to_additive]
+lemma semiconj_by.function_semiconj_mul_left (h : semiconj_by a b c) :
+  function.semiconj ((*)a) ((*)b) ((*)c) :=
+λ j, by rw [← mul_assoc, h.eq, mul_assoc]
+
+@[to_additive]
+lemma commute.function_commute_mul_left (h : commute a b) :
+  function.commute ((*)a) ((*)b) :=
+semiconj_by.function_semiconj_mul_left h
+
+@[to_additive]
+lemma semiconj_by.function_semiconj_mul_right_swap (h : semiconj_by a b c) :
+  function.semiconj (*a) (*c) (*b) :=
+λ j, by simp_rw [mul_assoc, ← h.eq]
+
+@[to_additive]
+lemma commute.function_commute_mul_right (h : commute a b) :
+  function.commute (*a) (*b) :=
+semiconj_by.function_semiconj_mul_right_swap h
```

Comment:
I'm a bit unsure about the position of these lemmas.

Reply:
I can understand this: I do not really know what they say!

Feel free to move them wherever you feel that they belong!

Reply:
I don't have a better suggestion. In some sense it fits, but I'm not sure this would be the place where I would look for them. I just wanted to flag it for reviewers.